rfcs/20200612-stream-executor-c-api.md

```diff
@@ -72,65 +72,123 @@ A decoupled way to add a new device to TensorFlow.
-is quite large and some of its methods are only sporadically used. Therefore, we
-plan to wrap only a subset of key StreamExecutorInterface functionality. We decided on this subset based on the PluggableDevice usecase as well as potential future devices such as TPUs.
-
-Implementation conventions:
-
-* Structs include `struct_size` parameter. This parameter should be filled in both by TensorFlow and the plugin and can be checked to determine which struct fields are available for current version of TensorFlow.
-* Struct name prefixes indicates which side of the API is responsible for populating the struct:
-  * `SE_` prefix: filled by TensorFlow.
-  * `SP_` prefix: filled by plugins, except `struct_size` which is also filled by TensorFlow when TensorFlow passes it to a callback.
-
-See proposed C API below:
-
-```cpp
-#include <stddef.h>
-#include <stdint.h>
-
+has a large number of methods, some of which are only sporadically used.
+Therefore, we plan to wrap only a subset of key `StreamExecutorInterface`
+functionality. We decided on this subset based on the [PluggableDevice](https://github.com/tensorflow/community/pull/262) 
+usecase as well as potential future devices such as TPUs.
+
+## Versioning Strategy and Stability
+StreamExecutor C API follows Semantic Versioning 2.0.0 ([semver](http://semver.org/)).
+Each release version has a format `MAJOR.MINOR.PATCH`, as outlined in
+[TensorFlow version compatibility](https://www.tensorflow.org/guide/versions#semantic_versioning_20).
+We also use struct sizes to track compatibility. More details on functionality
+extension and deprecation in [StreamExecutor C API Versioning Strategy](20200612-stream-executor-c-api/C_API_versioning_strategy.md).
+
+The C API will have an initial bake-in period, where we won’t have any
+compatibility guarantees. However, we will make the best effort to perform any
+updates in a backwards compatible way. For example, we plan to keep track of
+struct sizes. 
+
+The C API will be placed in [tensorflow/c/experimental](https://cs.opensource.google/tensorflow/tensorflow/+/refs/tags/v2.3.0:tensorflow/c/experimental/).
+We will consider moving the API out of the experimental directory once it is
+more stable.
+
+## Implementation Conventions
+
+* Struct prefix indicates whether struct fields should be filled by the plug-in or core implementation:
+  * `SE_`: Set/filled by core, unless marked otherwise.
+  * `SP_`: Set/filled by plug-in, unless marked otherwise.
+  * This prefix rule only applies to structures. Enumerations and methods are all prefixed with `SE_`.
+* Structs begin with two fields:
+  * `size_t struct_size`: Stores the unpadded size of the struct.
+  * `void* ext`: A free-form field that can be populated by a plugin in `SP_*` structs or potential future extension points in `SE_` structs.
+* We use `struct_size` for version checking.
+  * It is exempt from the `SE/SP` rule above and should be set both by core and plug-in.
```

Comment:
```suggestion
  * It is exempt from the `SE/SP` rule above and must be set both by core and plug-in.
```
RFC 2119

Reply:
Thank you for the detailed review, Situ! (I know you're not done yet. Just committing one batch of changes first.)

> RFC 2119

What do you mean by `RFC 2119`?